### ISSUE TITLE
Change icon size to em

### ISSUE THREAD

**Issue by steff456** [spyder-ide/spyder#15128]
The icons in big screens are not resized as expected making them to be too small for the toolbar

![image](https://user-images.githubusercontent.com/20992645/113638370-50a31000-963c-11eb-8bf7-1f4343d26773.png)

**Comment by mIgLLL** [comment]
I have the same problem.

**Comment by ccordoba12** [comment]
What's your screen resolution @mIgLLL?

**Comment by ccordoba12** [comment]
It seems we need to do this to solve the bug: https://stackoverflow.com/questions/55842276/how-to-resize-or-scale-a-qicon

**Comment by mIgLLL** [comment]
> What's your screen resolution @mIgLLL?

I don't have solution so I roll back to 4.2.3, and I am just reporting this bug. LOL

**Comment by leavor** [comment]
> > What's your screen resolution @mIgLLL?
> 
> I don't have solution so I roll back to 4.2.3, and I am just reporting this bug. LOL

哥，意思是问你的屏幕分辨率是多少，并不是问你解决方法是什么。。。

**Comment by mIgLLL** [comment]
> What's your screen resolution @mIgLLL?

3840x2160

**Comment by mIgLLL** [comment]
> > > What's your screen resolution @mIgLLL?
> > 
> > 
> > I don't have solution so I roll back to 4.2.3, and I am just reporting this bug. LOL
> 
> 哥，意思是问你的屏幕分辨率是多少，并不是问你解决方法是什么。。。

谢了

**Comment by dovsay** [comment]
![image](https://user-images.githubusercontent.com/36002974/113831614-63dbea00-97ba-11eb-9477-317c28347c0f.png)

**Comment by ccordoba12** [comment]
Thanks @leavor for your help and @mIgLLL for your answer!

**Comment by mrkwjc** [comment]
Same problem here. `Enable auto high DPI scaling` option is not working for me.

**Comment by Aegis1863** [comment]
> Same problem here. `Enable auto high DPI scaling` option is not working for me.

Me too :<

**Comment by ccaprani** [comment]
> Me too :<

Me three. Same issue; changing the DPI doesn't do anything.

**Comment by green-slowmoe** [comment]
same here. DPI scaling did the trick on 4.x, but not anymore

**Comment by 1900011604** [comment]
So why do you move the milestone from 5.0.2 all to 5.1.1?

**Comment by ccordoba12** [comment]
> So why do you move the milestone from 5.0.2 all to 5.1.1?

Because we didn't have time to implement this for previous versions.

> Is there any way for us to solve this problem like changing a setup file?

No, there isn't, sorry.

**Comment by 1900011604** [comment]
I found a temporary solution for this:

![Screenshot_20210922_121536](https://user-images.githubusercontent.com/65082432/134282833-3578acbf-8e08-4ecc-becf-362f30ddfd72.png)

```
set it to 1.0, it will restart but nothing changed, so:
close the console, then open the console and start spyder
```

This method will make the icons fit the words, but the buttons are still **E X T R E M E L Y  L A R G E**.

**Comment by ccordoba12** [comment]
What happens if instead you select the `Enable auto high DPI scaling` option instead?

**Comment by 1900011604** [comment]
> What happens if instead you select the `Enable auto high DPI scaling` option instead?

It returns to the wrong display:
![Screenshot_20210924_101233](https://user-images.githubusercontent.com/65082432/134608183-722ab7c0-eeb3-48f5-b468-26260dbd122b.png)

**Comment by ccordoba12** [comment]
Ok, what if you adjust the scaling factor to be 1.125?

**Comment by 1900011604** [comment]
It will be sized to 112.5%, both icon and text, also for any scaling factor > 1.0

**Comment by ccordoba12** [comment]
I know, but that could help to fix the scaling issues with buttons.

**Comment by Firestar-Reimu** [comment]
> It seems we need to do this to solve the bug: https://stackoverflow.com/questions/55842276/how-to-resize-or-scale-a-qicon

So you cannot change the iconSize property of QToolBar?

**Comment by ccordoba12** [comment]
It is possible, but we haven't had time to take a look at that, sorry.

**Comment by MelvaMud** [comment]
Hi, 

Will the resizing of the icons in Spyder 5.1.5 ever be addressed? I was able to resize once, but it's reverted back to such tiny letter it's impossible to read the icons unless I use a jeweler's loupe. Just ridiculous.

Thanks

**Comment by Firestar-Reimu** [comment]
Today test on 5.3.1:
![Screenshot_20220606_132507](https://user-images.githubusercontent.com/65082432/172111954-0420d625-85e8-48ec-896a-452cb804cafe.png)

**Comment by frispete** [comment]
Hi, 5.4.2 with current Linux Tumbleweed on a 16" display with 3840x2400:
Custom scale factor: 1.2
![grafik](https://user-images.githubusercontent.com/1009685/221411432-096f4802-9c9c-47b1-9fb8-aedaee260064.png)
Best result with `QT_AUTO_SCREEN_SCALE_FACTOR=1`, but still wastes a lot of space around the icons:
![grafik](https://user-images.githubusercontent.com/1009685/221412156-b80e9fa0-d905-4df8-ad3e-23ce5436a399.png)

**Comment by Firestar-Reimu** [comment]
> but still wastes a lot of space around the icons

it seems to be a feature

**Comment by ccordoba12** [comment]
Yes, it is feature to comply with accessibility standards: buttons need to have a minimal size in order to be accesible, and that's the size we're using in Spyder.

**Comment by BYUJason** [comment]
I fixed this somehow but then I installed a new driver and it's back to having huge icon and menu text and no setting change will fix it. I tried installing the newest version 5.4.3, it did not fix.

Is there a solution in the pipeline yet?

**Comment by ccordoba12** [comment]
I think this will be fixed in Spyder 6 because we switch from setting buttons' size in ems to pixels.
